stdlib/REPL/src/REPL.jl

```diff
@@ -691,7 +691,7 @@ function return_callback(s)
 end
 
 find_hist_file() = get(ENV, "JULIA_HISTORY",
-    joinpath(homedir(), ".julia", "logs", "repl_history.jl"))
+    !isempty(DEPOT_PATH) ? joinpath(DEPOT_PATH[1], "logs", "repl_history.jl") : "")
```

Comment:
Maybe return `nothing` instead of empty string and check for that explicitly at
https://github.com/JuliaLang/julia/blob/fd17073de8336cb9d2ca5a2abad45cee230fde0c/stdlib/REPL/src/REPL.jl#L851

https://github.com/JuliaLang/julia/blob/fd17073de8336cb9d2ca5a2abad45cee230fde0c/stdlib/REPL/src/REPL.jl#L852 might be bad if we return `""`.